elements/lisk-p2p/src/p2p.ts

```diff
@@ -257,8 +263,13 @@ export class P2P extends EventEmitter {
 			sanitizedPeerLists: this._sanitizedPeerLists,
 			secret: this._secret,
 		});
-		codec.addSchema(peerInfoSchema);
-		codec.addSchema(nodeInfoSchema);
+
+		this._rpcSchema = {
+			peerInfo: config.customSchema?.peerInfo ?? peerInfoSchema,
+			nodeInfo: config.customSchema?.nodeInfo ?? nodeInfoSchema,
+		};
+		codec.addSchema(this._rpcSchema.peerInfo);
```

Comment:
It would `_handlePeerPoolRPC` is a callback that's registered for actions if that's the case maybe this `addSchema` calls should happen somewhere else as schema caching is expensive so we want to use it as little as possible.

Reply:
This `addSchema` is in the constructor of P2P(not inside `_handlePeerPoolRPC`) and only called once on startup, if this is what you meant

Reply:
Oh my parser failed to get the correct scope. Sorry!